pkg/cli/admin/nodeimage/create.go

```diff
@@ -331,20 +337,64 @@ func (o *CreateOptions) copyArtifactsFromNodeJoinerPod() error {
 	klog.V(2).Infof("Copying artifacts from %s", o.nodeJoinerPod.GetName())
 	rsyncOptions := &rsync.RsyncOptions{
 		Namespace:     o.nodeJoinerNamespace.GetName(),
-		Source:        &rsync.PathSpec{PodName: o.nodeJoinerPod.GetName(), Path: path.Join("/assets", "node.x86_64.iso")},
+		Source:        &rsync.PathSpec{PodName: o.nodeJoinerPod.GetName(), Path: "/assets/"},
 		ContainerName: nodeJoinerContainer,
-		Destination:   &rsync.PathSpec{PodName: "", Path: path.Join(o.AssetsDir, o.OutputName)},
+		Destination:   &rsync.PathSpec{PodName: "", Path: o.AssetsDir},
 		Client:        o.Client,
 		Config:        o.Config,
 		Compress:      true,
 		RshCmd:        fmt.Sprintf("%s --namespace=%s -c %s", o.rsyncRshCmd, o.nodeJoinerNamespace.GetName(), nodeJoinerContainer),
 		IOStreams:     o.IOStreams,
 		Quiet:         true,
+		RsyncInclude:  []string{"*.iso"},
+		RsyncExclude:  []string{"*"},
 	}
 	rsyncOptions.Strategy = o.copyStrategy(rsyncOptions)
 	return rsyncOptions.RunRsync()
 }
 
+func (o *CreateOptions) renameImageIfOutputNameIsSpecified() error {
+	if o.OutputName == "" {
+		return nil
+	}
+	// AssetDir doesn't exist in unit test fake filesystem
+	_, err := os.Stat(o.AssetsDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		} else {
+			return err
+		}
+	}
+
+	err = filepath.Walk(o.AssetsDir, func(path string, info os.FileInfo, err error) error {
```

Comment:
wondering if in this case using [os.ReadDir](https://pkg.go.dev/os#ReadDir) could lead to a simpler implementation

Reply:
I tried os.ReadDir. It is pretty much the same.